### ISSUE TITLE
RequestLocateNameReq for fields sends incorrect result

### ISSUE THREAD

**Issue by chipsenkbeil** [ensime/ensime-server#1508]
Not sure how I let this one get passed me. The following is a snippet from the `DebugLocateNameReq`:

```scala
            tr.get.flatMap {
              case v: IndexedVariableInfoProfile =>
                println(s"Found local variable for $name")
                Some(DebugStackSlot(DebugThreadId(t.cache().uniqueId), v.frameIndex, v.offsetIndex))
              case v if v.isField => v.toValueInfo match {
                case o if o.isObject =>
                  println(s"Found object field for $name")
                  val oo = o.toObjectInfo
                  Some(DebugObjectField(DebugObjectId(oo.cache().uniqueId), v.name))
                case _ =>
                  println(s"Found primitive field for $name")
                  None
              }
```

The field logic is entirely wrong. What it should be returning is `Some(DebugObjectField(<THIS OBJECT CONTAINING FIELD>, v.name))`. Instead, it is returning the field if it is an object and `None` otherwise. Instead, the field needs to have a reference to its containing parent (it already does internally) and access that.

This is going to require a change to the debugger api to have a standard method to return the parent object for the field (and will require a specific trait to expose it). Not hard, but going to have to publish an M2 once this (and any other required changes for the 1.0 release) are finished.

**Comment by rorygraves** [comment]
M2 makes sense - better to catch it now rather than later - well done @chipsenkbeil @dickwall

**Comment by chipsenkbeil** [comment]
FYI, fixed the ensime server code with using an update in the debugger api. Now able to successfully retrieve object fields that have primitive and object values. Not going to close until the code itself has been published as 1.1.0-M2 for the debugger api and integrated into ensime, of course.

**Comment by chipsenkbeil** [comment]
Resolver didn't seem to work in the PR. Fixed via #1527.
